state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1180,30 +1192,82 @@ pub mod pallet {
 			Ok(())
 		}
 
-		/// Associates `short_id` with `affiliate_id` for a given broker. Overwrites the record
-		/// under `short_id` if already taken by another affiliate.
+		/// Registers an affiliate for a broker. Sets the withdrawal address for the affiliate which
+		/// will be used to withdraw the affiliate's fees. The withdrawal address can not be changed
+		/// once set. The broker will be able to trigger a withdrawal
+		/// request to the affiliate's withdrawal address. A short id is taken and can only be used
+		/// at a time. The affiliate account is derived from the broker account id.
+		///
+		/// ## Events
+		///
+		/// - [AffiliateRegistration](Event::AffiliateRegistration)
 		#[pallet::call_index(14)]
 		#[pallet::weight(T::WeightInfo::register_affiliate())]
 		pub fn register_affiliate(
 			origin: OriginFor<T>,
-			affiliate_id: T::AccountId,
 			short_id: AffiliateShortId,
+			withdrawal_address: EncodedAddress,
 		) -> DispatchResult {
 			let broker_id = T::AccountRoleRegistry::ensure_broker(origin)?;
 
-			let previous_affiliate_id = AffiliateIdMapping::<T>::take(&broker_id, short_id);
+			ensure!(
+				!AffiliateIdMapping::<T>::contains_key(&broker_id, short_id),
+				Error::<T>::AffiliateAlreadyRegistered
+			);
+
+			ensure!(
+				T::AddressConverter::decode_and_validate_address_for_asset(
+					withdrawal_address.clone(),
+					Asset::Eth,
```

Comment:
It doesn't make much difference but actually the asset is Usdc. 

We could also require the address to be an `EthereumAddress`, then we don't need to do this check.